services/mailer/mailer.go

```diff
@@ -61,10 +101,13 @@ func (m *Message) ToMessage() *gomail.Message {
 		if strings.Contains(base.TruncateString(m.Body, 100), "<html>") {
 			log.Warn("Mail contains HTML but configured to send as plain text.")
 		}
-		msg.SetBody("text/plain", plainBody)
+		msg.SetBody("text/plain", clearSignedMessage(plainBody))
 	} else {
-		msg.SetBody("text/plain", plainBody)
+		msg.SetBody("text/plain", clearSignedMessage(plainBody))
 		msg.AddAlternative("text/html", m.Body)
+		if setting.MailService.SignEmails {
+			msg.AddAlternative("application/pgp-signature", detachedSignature(m.Body))
```

Comment:
The HTML version is not included in the signature. What does the user agent do in that case? Skip the HTML? 🤔

Reply:
The html block is signed with a detached signature

Reply:
I don't think the there's a way of signing both with one signature hence the decision to sign the plaintext in clear sign and the html as a detached signature. That being said if you know what the best practice is I'm only guessing here.